OutOfSchool/OutOfSchool.WebApi/Services/ParentService.cs

```diff
@@ -86,6 +87,37 @@ public async Task<IEnumerable<ParentDTO>> GetAll()
             return parents.Select(parent => parent.ToModel()).ToList();
         }
 
+        /// <inheritdoc/>
+        public async Task<SearchResult<ParentDTO>> GetByFilter(SearchStringFilter filter)
+        {
+            logger.LogInformation("Getting all Parents started (by filter).");
+
+            if (filter is null)
+            {
+                filter = new SearchStringFilter();
+            }
+
+            var filterPredicate = PredicateBuild(filter);
+
+            int count = await repositoryParent.Count(filterPredicate).ConfigureAwait(false);
```

Comment:
Why do we need 2 calls to the data source ?
Correct me if I'm wrong but the code below gets the collection of items from the data source by the same filter.
Is that the same count ?
Is it possible to get this count from items in the collection ?